### ISSUE TITLE
Incorrect string parsing of QuadDouble

### ISSUE THREAD

**Issue by Andreas113** [neslib/Neslib.MultiPrecision#8]
Hi Erik,
have just discovered a small malfunction in the **parser** routine of **QuadDouble**. 

```
Here is an example:

Procedure Print(Name: String; Value: QuadDouble);
VAR
  S: String;
Begin
   S:= QuadDouble.ToString(Value, TMPFloatFormat.Fixed);
   WriteLn(Name + ' = ', S);
   WriteLn;
End;{Print}
{---------}

VAR
  ZQd: QuadDouble;
  St : String;
. . . 
St:= '12345.678901234567890123456789012345678901234567890123456789012345678901';
ZQd:= St;
Print('ZQd: Only the first 33 figures are correct: ', ZQd);

// ---> Result:
ZQd: Only the first 33 figures are correct:  = 12345.67890123456789012345678901235113471715242452040893925930542385
```
-->  12345.6789012345678901234567890123**5113471715242452040893925930542385**

Could you please have a look at it if you have time?
Thank you!

Regards,
Andreas

**Comment by neslib** [comment]
Andreas,

I cannot reproduce this issue with Delphi 10.4. When I run this code, the result is accurate to at least 64 digits. I tried 32-bit and 64-bit Windows, both with and without the MP_ACCURATE define.

What Delphi version did you use? Did you make any modifications to the source code?

**Comment by Andreas113** [comment]
Hi Erik,
in the meantime, I was able to locate and find out the reason for my erroneous result above: 
I had only made one change to your code before your last update, which caused the error:

In
`  DoubleDouble = record`

I added the following:

```
    class operator Implicit(const Value: Extended): DoubleDouble; inline; static;
      ...
    class operator DoubleDouble.Implicit(const Value: Extended): DoubleDouble;

    VAR
      a, b: Double;
      a:= Value;

    Begin
      b:= Value - a;

      Result.X[0]:= a + b;
      Result.X[1]:= b - (Result.X[0] - a);
    end;
```
Strange, because the error **occurred in QuadDouble**! This **DoubleDouble.Implicit(..)** was actually not called at all and yet this was the cause of the error. I guess it's an unexpected side effect. 
I can only praise your foresight and experience, because you consistently do not use Implicit(..), but always only Explicit(..). Now I know very well why you do it that way. 

**You are an excellent programmer, Erik!**  I have learned a lot from you again! 

Thank you!
Andreas
PS:
We can close this topic, sorry for my false alarm....
